server/src/test/java/io/druid/server/coordinator/DruidCoordinatorTest.java

```diff
@@ -317,8 +326,16 @@ public void childEvent(
     catch (Exception ex) {
       throw ex;
     }
-    Assert.assertTrue(assignSegmentLatch.await(4, TimeUnit.SECONDS));
+
+    Assert.assertTrue(assignSegmentLatch.await(2, TimeUnit.SECONDS));
+    Assert.assertEquals(ImmutableMap.of(dataSource, 100.0), coordinator.getLoadStatus());
     curator.delete().guaranteed().forPath(ZKPaths.makePath(LOADPATH, dataSegment.getIdentifier()));
+    // Wait for coordinator thread to run so that replication status is updated
+    // Waiting 50 ms more than the Coordination period should be sufficient
+    Thread.sleep(COORDINATOR_PERIOD + 50);
```

Comment:
This wait is for the coordinator thread to run and update its view of the replication status in the cluster so I believe that latches cannot be used. However, waiting in while loop can be done.